physical/raft/raft.go

```diff
@@ -489,9 +489,22 @@ func (b *RaftBackend) CollectMetrics(sink *metricsutil.ClusterMetricSink) {
 	b.l.RLock()
 	logstoreStats := b.stableStore.(*raftboltdb.BoltStore).Stats()
 	fsmStats := b.fsm.db.Stats()
+	stats := b.raft.Stats()
 	b.l.RUnlock()
 	b.collectMetricsWithStats(logstoreStats, sink, "logstore")
 	b.collectMetricsWithStats(fsmStats, sink, "fsm")
+	labels := []metrics.Label{
+		{
+			Name:  "peer_id",
+			Value: b.localID,
+		},
+	}
+	for _, key := range []string{"term", "commit_index", "applied_index", "fsm_pending"} {
```

Comment:
One small thing here. It's not super obvious, but the `b.raft.Stats()["applied_index"]` is actually the latest index that the raft library has *queued* to the FSM (it includes `fsm_pending` items), not what has actually been applied. Our heartbeating mechanism uses the actual last applied index we've seen in the FSM. If the intent is to compare this value between nodes you might find that it's more up-to-date then reality. If that's not the intent (which maybe not given the existence of the delta metric below), then we may just want to leave it as is and mention this fact in the docs?

Reply:
I think I'll take the doc strategy.  As you say, applied_index_delta can tell us how far behind followers are.  IIRC here I was mostly just looking to expose potentially useful info from the raft lib for debugging.